sdk/src/Core/Amazon.Runtime/CredentialManagement/AWSCredentialsFactory.cs

```diff
@@ -302,7 +309,11 @@ private static AWSCredentials GetAWSCredentialsInternal(
 
                         roleSessionName = options.RoleSessionName ?? RoleSessionNamePrefix + AWSSDKUtils.CorrectedUtcNow.Ticks;
                         assumeRoleOptions = new AssumeRoleAWSCredentialsOptions();
-                        return new AssumeRoleAWSCredentials(sourceCredentials, options.RoleArn, roleSessionName, assumeRoleOptions);
+                        return new AssumeRoleAWSCredentials(sourceCredentials, options.RoleArn, roleSessionName, assumeRoleOptions)
+                        {
+                            FeatureIdSource = UserAgentFeatureId.CREDENTIALS_PROFILE_NAMED_PROVIDER,
```

Comment:
These all seem like they should have the `CREDENTIALS_STS_ASSUME_ROLE` feature id but I can also see that it is named profile so an argument can be made for `CREDENTIALS_PROFILE_NAMED_PROVIDER`. Does `FeatureIdSource` need to be a collection?

Reply:
Just checked the JS implementation and they do add multiple feature IDs. I tested with a source profile and their user agent had both `CREDENTIALS_STS_ASSUME_ROLE` and `CREDENTIALS_PROFILE_SOURCE_PROFILE`:
```
[profile A]
role_arn = arn:aws:iam::000000000000:role/test-assume-role
credential_source = Environment
```

Updated the feature ID source property to be a collection in https://github.com/aws/aws-sdk-net/pull/3717/commits/711713677aaa84fb87f9bcb9a2f92146ba29585f